impl/maven-core/src/main/java/org/apache/maven/configuration/internal/EnhancedCompositeBeanHelper.java

```diff
@@ -304,20 +301,17 @@ private Object convertProperty(
      * Set field value with cached accessibility.
      */
     private void setFieldValue(Object bean, Field field, Object value) throws IllegalAccessException {
-        Boolean isAccessible = ACCESSIBLE_FIELD_CACHE.get(field);
-        if (isAccessible == null) {
-            isAccessible = field.canAccess(bean);
-            if (!isAccessible) {
-                field.setAccessible(true);
-                isAccessible = true;
-            }
-            ACCESSIBLE_FIELD_CACHE.put(field, isAccessible);
-        } else if (!isAccessible) {
+        boolean wasAccessible = field.canAccess(bean);
+        if (!wasAccessible) {
             field.setAccessible(true);
```

Comment:
This code assumes that the same `Field` object will not be set by two threads in same time. It seems to imply that `EnhancedCompositeBeanHelper` is not really thread-safe. If this limitation okay?